service/history/transfer_queue_active_task_executor.go

```diff
@@ -216,13 +216,14 @@ func (t *transferQueueActiveTaskExecutor) processActivityTask(
 
 	timeout := timestamp.DurationValue(ai.ScheduleToStartTimeout)
 	directive := MakeDirectiveForActivityTask(mutableState, ai)
+	wfVersioningInfo := workflow.GetWorkflowVersioningInfoMatchingTask(mutableState)
```

Comment:
see this scare NOTE below.

```suggestion
	wfVersioningInfo := common.CloneProto(workflow.GetWorkflowVersioningInfoMatchingTask(mutableState))
```

Reply:
removed this